src/Elements/Button/Button.tsx

```diff
@@ -0,0 +1,97 @@
+import cx from 'classnames';
+import * as React from 'react';
+
+//Style
+import './button.scss';
+
+// types
+interface IProps {
+  /**
+   * class attribute of the component
+   */
+  className?: string;
+  /**
+   * content of the component
+   */
+  children?: React.ReactNode;
+  /**
+   * declare if the button is visible or not
+   */
+  visible?: boolean;
+  /**
+   * icon to be shown on the button
+   */
+  icon?: string;
+  /**
+   * icon positions
+   */
+  iconPosition?: 'left' | 'right';
+  /**
+   * if the button is disabled or not
+   */
+  disabled?: boolean;
+  /**
+   * label of the component
+   */
+  label?: string;
+  /**
+   * position of the label
+   */
+  labelPosition?: 'right' | 'left'
+  /**
+   * on click event handler function
+   */
+  onClick?: Function;
```

Comment:
the click event handler should be required, I can't see any use case for a button without onClick handler